var/spack/repos/builtin/packages/perl-uri/package.py

```diff
@@ -13,5 +13,4 @@ class PerlUri(PerlPackage):
     url      = "http://search.cpan.org/CPAN/authors/id/E/ET/ETHER/URI-1.72.tar.gz"
 
     version('1.72', 'cd56d81ed429efaa97e7f3ff08851b48')
-
-    depends_on('perl-test-needs', type=('build', 'run'))
```

Comment:
I believe this dependency should be kept, but the type should be `'test'`.

Reply:
Okay; I put the dependency back in with `type=test`. This will be the only perl package that has a type=test dependency and I am guessing that there would be quite a few more perl packages that could have type=test dependencies. I had to set the dependency type to `('build', 'test')` in order for tests to succeed.

Reply:
I wonder if there is a bug in spack somewhere. If `depends_on('perl-test-needs', type='test'):
```console
$ spack install --test=root perl-uri
...
==> Installing perl-test-needs
==> Searching for binary cache of perl-test-needs
==> Finding buildcaches in /nfs/install/spack/argon_mirror/build_cache
==> No binary for perl-test-needs found: installing from source
==> Using cached archive: /opt/packages/gpjohnsn/opt/ssoft/spack/var/spack/cache/perl-test-needs/perl-test-needs-0.002005.tar.gz
==> Staging archive: /tmp/gpjohnsn/spack-stage/perl-test-needs-0.002005-tfbalkc33dxejeouiqhy6wns57id7nr7/Test-Needs-0.002005.tar.gz
==> Created stage in /tmp/gpjohnsn/spack-stage/perl-test-needs-0.002005-tfbalkc33dxejeouiqhy6wns57id7nr7
==> No patches needed for perl-test-needs
==> Building perl-test-needs [PerlPackage]
==> Executing phase: 'configure'
==> Executing phase: 'build'
==> Executing phase: 'install'
==> Successfully installed perl-test-needs
  Fetch: 0.00s.  Build: 0.81s.  Total: 0.81s.
[+] /opt/packages/gpjohnsn/opt/ssoft/spack/opt/spack/linux-centos7-haswell/gcc-4.8.5/perl-test-needs-0.002005-tfbalkc33dxejeouiqhy6wns57id7nr7
==> Installing perl-uri
==> Searching for binary cache of perl-uri
==> Finding buildcaches in /nfs/install/spack/argon_mirror/build_cache
==> No binary for perl-uri found: installing from source
==> Using cached archive: /opt/packages/gpjohnsn/opt/ssoft/spack/var/spack/cache/perl-uri/perl-uri-1.72.tar.gz
==> Staging archive: /tmp/gpjohnsn/spack-stage/perl-uri-1.72-ltkwl4qzhrkzpktequzavjuricouufbx/URI-1.72.tar.gz
==> Created stage in /tmp/gpjohnsn/spack-stage/perl-uri-1.72-ltkwl4qzhrkzpktequzavjuricouufbx
==> No patches needed for perl-uri
==> Building perl-uri [PerlPackage]
==> Executing phase: 'configure'
==> Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16' 'test'

1 error found in build log:
     192    t/urn-isbn.t            (Wstat: 512 Tests: 0 Failed: 0)
     193      Non-zero exit status: 2
     194      Parse errors: No plan found in TAP output
     195    Files=43, Tests=622,  2 wallclock secs ( 0.11 usr  0.02 sys +  1.16
             cusr  0.20 csys =  1.49 CPU)
     196    Result: FAIL
     197    Failed 2/43 test programs. 0/622 subtests failed.
  >> 198    make: *** [test_dynamic] Error 255

See build log for details:
  /tmp/gpjohnsn/spack-stage/perl-uri-1.72-ltkwl4qzhrkzpktequzavjuricouufbx/spack-build-out.txt
```
```
# *** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***
#
# The following REQUIRED prerequisites were not satisfied:
#
# Test::Needs is not installed (any version required)
#
```

The `Test::Needs.pm` module is not loaded into the spack environment so the tests fail.
```
Can't locate Test/Needs.pm in @INC (you may need to install the Test::Needs module) (@INC contains:...
```
Setting `depends_on('perl-test-needs', type=('build', 'test')) works as then the Test/Needs module is loaded in the spack environment.